src/components/container/SurveyTaskList/SurveyTaskList.tsx

```diff
@@ -136,6 +144,10 @@ export default function (props: SurveyTaskListProps) {
 		return null;
 	}
 
+	if (props.status == "incomplete" && !tasks?.length && props.hideIfEmpty) {
+		return null;
+	}
+
```

Comment:
The combination here made me double check the `SurveyTaskStatus` type, which then begs the question of what the proper behavior is if `status` is `closed`.  Admittedly, it's probably an infrequent case - but I feel like `complete` is a somewhat rare case too?  And the fact that the list *always* hides if you're using `complete` is probably a largely unknown functionality?  I think it's confusing that 2 of the 3 statuses would basically ignore this property though.  This sort of interaction is largely hidden and hard to document.

Maybe it's simply worth collapsing both of these cases down to remove `status` as a variable at all with the addition of the `hideIfEmpty` property?  There would be a slight chance of behavior change if someone upgraded and was using `complete` because `hideIfEmpty` is technically `false` if it isn't set.  We could either accept that risk, or we could swap this to `showIfEmpty` (though I think that's a little more confusing, potentially).

Reply:
In terms of rendering the task list, `closed` is not really a valid [List Type](https://github.com/CareEvolution/MyDataHelpsViewBuilder/blob/main/src/nodes/SurveyTaskListNode.tsx#L71-L74), as far as View Builder is concerned [1]. So maybe `status` should not be typed to `SurveyTaskStatus` since it includes `closed`. The problem is that it's used as a [task query parameter](https://github.com/CareEvolution/MyDataHelpsUI/pull/322/files#diff-f70c12f89bdafafbba64c0a348a060fd6d05ce450e3f3dec972822eb63026264R102). Otherwise I'd be tempted to rename/redefine that property as `listType: "incomplete" | "complete";`.

I would say `complete` is commonly used. For EMPOWER, we started without a completed tasks list (I think it adds a bit of noise to the UI), but then site coordinators that did not have access to MDHD had no way of confirming whether the participant had completed their ePROs (surveys) before, e.g., getting their lab work done. AoU also had a completed tasks section, of course.

Your point on the asymmetry between `incomplete` and `complete` having different hidden-or-not semantics still stands, though. I wouldn't say "the fact that the list always hides if you're using complete is probably a largely unknown functionality", as that should (in theory) become readily apparent the moment you test your view (if building with View Builder). It is certainly opinionated to ignore the new `hideIfEmpty` prop if your list type is `complete`, so we could arguably honor the prop, if set, as long as the default behavior is to hide if empty, to ensure backwards compatibility. I guess if you put a completed task list in its own view, having a message that says "No tasks have been completed." makes sense. But if you have an incomplete task list next to it, it's kind of unnecessary since you have the incomplete tasks staring at you.

Incidentally, there is a [`SurveyTasksView`](https://ui.mydatahelps.org/?path=/story/view-surveytasksview--hide-incomplete) that implements its own hiding mechanism, which I should probably update.

[1] Which makes sense in terms of participant-facing use cases, at least.

Reply:
Regarding the `closed` status, I guess making this change wouldn't be too bad:
```
listType: "incomplete" | "complete";
...
var parameters: SurveyTaskQueryParameters = { status: props.listType }
```
But then we'd really be saying "do not use this component for closed tasks."

Reply:
Ok, see latest commit.